### PR TITLE
Add default context

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Adds `DPCTLPlatform_GetDefaultContext` that exposes ``sycl::platform::ext_oneapi_get_default_context()`` ([ext_oneapi_default_context](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_default_context.asciidoc) extension).

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
@diptorupd Do you think `SyclDevice.platform` should be `SyclDevice.sycl_platform` instead?

File: dpctl/_sycl_device.pyx [hunk 1]
```diff
@@ -639,6 +643,22 @@ cdef class SyclDevice(_SyclDevice):
...
+    def platform(self):
+        """ Returns the platform associated with this device.
+
+        Returns:
+            :class:`dpctl.SyclPlatform`: The platform associated with this
+                device.
+        """
+        cdef DPCTLSyclPlatformRef PRef = (
+            DPCTLDevice_GetPlatform(self._device_ref)
+            )
```
Comment (diptorupd) [inline dpctl/_sycl_device.pyx:656]: indent is off. Should line up with the `cdef`

File: dpctl/_sycl_device.pyx [hunk 2]
```diff
@@ -639,6 +643,22 @@ cdef class SyclDevice(_SyclDevice):
...
+        if (PRef == NULL):
+            raise RuntimeError("Could not get platform for device.")
+        else:
+            return SyclPlatform._create(PRef)
```
Comment (diptorupd) [inline dpctl/_sycl_device.pyx:660]: I am assuming `_create` will consume the `PRef`?

Comment (oleksandr-pavlyk) [reply dpctl/_sycl_device.pyx:660]: Yes, it does.

File: dpctl/_sycl_device.pyx (hunk 1 above)
Comment (oleksandr-pavlyk) [reply dpctl/_sycl_device.pyx]: Fixed

File: dpctl/tests/test_sycl_platform.py [hunk 3]
```diff
@@ -87,6 +87,11 @@ def check_repr(platform):
     assert r != ""
 
 
+def check_default_context(platform):
+    r = platform.default_context
+    assert type(r) is dpctl.SyclContext
```
Comment (diptorupd) [inline dpctl/tests/test_sycl_platform.py:92]: Do we ever expect `r` to be None?

Comment (diptorupd) [reply dpctl/tests/test_sycl_platform.py:92]: Guess not, but can run into an Exception. The test case can handle that?

Comment (oleksandr-pavlyk) [reply dpctl/tests/test_sycl_platform.py:92]: Yes, an exception would get raised and pytest will report it

File: libsyclinterface/source/dpctl_sycl_platform_interface.cpp [hunk 4]
```diff
@@ -202,3 +203,18 @@ __dpctl_give DPCTLPlatformVectorRef DPCTLPlatform_GetPlatforms()
...
+__dpctl_give DPCTLSyclContextRef
+DPCTLPlatform_GetDefaultContext(__dpctl_keep const DPCTLSyclPlatformRef PRef)
+{
+    auto P = unwrap(PRef);
+    if (P) {
+        auto default_ctx = P->ext_oneapi_get_default_context();
+        return wrap(new context(default_ctx));
+    }
+    else {
+        error_handler("Driver version cannot be looked up for a NULL platform.",
```
Comment (diptorupd) [inline libsyclinterface/source/dpctl_sycl_platform_interface.cpp:216]: Hmm what is this error message? Copy pasta?

Comment (oleksandr-pavlyk) [reply libsyclinterface/source/dpctl_sycl_platform_interface.cpp:216]: Right. Sorry.

File: libsyclinterface/tests/test_sycl_queue_interface.cpp [hunk 5]
```diff
@@ -446,7 +446,6 @@ TEST_P(TestDPCTLQueueMemberFunctions, CheckMemset)
 
     ASSERT_NO_FATAL_FAILURE(DPCTLfree_with_queue(p, QRef));
 
-    bool equal = true;
```
Comment (diptorupd) [inline libsyclinterface/tests/test_sycl_queue_interface.cpp:449]: unrelated?

File: dpctl/tests/test_sycl_device.py [hunk 6]
```diff
@@ -552,6 +557,8 @@ def check_profiling_timer_resolution(device):
     check_repr,
     check_get_global_mem_size,
     check_get_local_mem_size,
+    check_profiling_timer_resolution,
```
Comment (diptorupd) [inline dpctl/tests/test_sycl_device.py:560]: unrelated?

File: libsyclinterface/tests/test_sycl_queue_interface.cpp (hunk 5 above)
Comment (oleksandr-pavlyk) [reply libsyclinterface/tests/test_sycl_queue_interface.cpp:449]: Yes, and in a separate commit.

File: dpctl/tests/test_sycl_device.py (hunk 6 above)
Comment (oleksandr-pavlyk) [reply dpctl/tests/test_sycl_device.py:560]: Yes, I realized this check added earlier was not exercised, so added it along with the `check_platform`

**Review by diptorupd — APPROVED** [review]
